File: src/test/com/huhx0015/algorithmtester/algorithms/BinaryTreeTraversalTest.java
package test.com.huhx0015.algorithmtester.algorithms;

import main.com.huhx0015.algorithmtester.datastructures.Node;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import static main.com.huhx0015.algorithmtester.algorithms.BinaryTreeTraversal.inOrderTraversal;
import static main.com.huhx0015.algorithmtester.algorithms.BinaryTreeTraversal.postOrderTraversal;
import static main.com.huhx0015.algorithmtester.algorithms.BinaryTreeTraversal.preOrderTraversal;

/**
 * Created by Michael Yoon Huh on 6/9/2017.
 */
class BinaryTreeTraversalTest {

    private static Node mTestNode;

    @BeforeAll
    static void initTestNode() {
        mTestNode = new Node(8);
        Node node1 = new Node(1);
        Node node2 = new Node(3);
        Node node3 = new Node(5);
        Node node4 = new Node(7);
        Node node5 = new Node(10);
        Node node6 = new Node(12);
        Node node7 = new Node(15);
        Node node8 = new Node(20);

        // LEFT SIDE
        mTestNode.left = node3;
        node3.left = node2;
        node3.right = node4;
        node2.left = node1;

        // RIGHT SIDE
        mTestNode.right = node6;
        node6.left = node5;
        node6.right = node7;
        node7.right = node8;
    }

    @Test
    void testInOrderTraversal() {
        inOrderTraversal(mTestNode);
    }

    @Test
    void testPreOrderTraversal() {
        preOrderTraversal(mTestNode);
    }

    @Test
    void testPostOrderTraversal() {
        postOrderTraversal(mTestNode);
    }
}

File: src/test/com/huhx0015/algorithmtester/2017/algorithms/BinaryTreeTraversalTest.kt
package test.com.huhx0015.algorithmtester.algorithms

import main.com.huhx0015.algorithmtester.datastructures.Node
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import main.com.huhx0015.algorithmtester.algorithms.BinaryTreeTraversal.inOrderTraversal
import main.com.huhx0015.algorithmtester.algorithms.BinaryTreeTraversal.postOrderTraversal
import main.com.huhx0015.algorithmtester.algorithms.BinaryTreeTraversal.preOrderTraversal

/**
 * Created by Michael Yoon Huh on 6/9/2017.
 */
internal class BinaryTreeTraversalTest {

    private var mTestNode: Node? = null

    @BeforeEach
    fun initTestNode() {
        mTestNode = Node(8)
        val node1 = Node(1)
        val node2 = Node(3)
        val node3 = Node(5)
        val node4 = Node(7)
        val node5 = Node(10)
        val node6 = Node(12)
        val node7 = Node(15)
        val node8 = Node(20)

        // LEFT SIDE
        mTestNode!!.left = node3
        node3.left = node2
        node3.right = node4
        node2.left = node1

        // RIGHT SIDE
        mTestNode!!.right = node6
        node6.left = node5
        node6.right = node7
        node7.right = node8
    }

    @Test
    fun testInOrderTraversal() {
        inOrderTraversal(mTestNode)
    }

    @Test
    fun testPreOrderTraversal() {
        preOrderTraversal(mTestNode)
    }

    @Test
    fun testPostOrderTraversal() {
        postOrderTraversal(mTestNode)
    }
}